### ISSUE TITLE
Improve text display

### ISSUE THREAD

**Issue by GlenHertz** [FugroRoames/TypedTables.jl#24]
I really like this table design as it is very Julian and simple to use, yet powerful -- nice work.  

Do you have any plans for an improved `show` method?  I would like to be able to do:

1. Show compact vs verbose view (with table borders)
2. Show all columns (wrapping around like `show(dataframe, true)`)
3. Have a way to format columns (ideally I could have a column displayed with the decimals '.' all aligned and scaled to the same units).  This is to make it easy to visually compare the size of the numbers in a vertical column.
4. Highlighting/bolding certain cells.

Assuming the Tables.jl API catches on then perhaps the text display features could go in another package for all implementations of the Table API.

**Comment by andyferris** [comment]
Yes we need this. I like those features but we may have to iterate to get there. 

And yeah, I was hoping that within weeks there will be a shared, third party tabular display. Perhaps i’m too optimistic. 

Otherwise we can take the code from MinimumViableTables or another package.

**Comment by andyferris** [comment]
@GlenHertz were you still interested in taking a stab at this, or should I proceed to implement something?

**Comment by GlenHertz** [comment]
@andyferris I ported the DataFrames display code over, see here:

https://github.com/GlenHertz/TypedTables.jl/commits/master

I'm not a huge fan of it since it is just a port -- don't feel obligated to take it.  

Initially I wanted to make it so it was easy to extend how a user wants to print a table.  I found that I can do something like `using Printf; TypedTables.ourshowcompact(io::IO, x::Float64) = @printf(io, "%.7f", ifelse(1e-20<abs(x)<1e-6, x*1e12, x))` to scale small floats and have the decimals line up.  This served me pretty well but I would like to be able to say that a column of a particular name and type should be printed in a specific way.  Currently the justification (left or right) is tied into other code which makes it hard to customize.
